### ISSUE TITLE
Getting Uncaught (in promise) TypeError: Cannot read property 'segmentAbove' of null

### ISSUE THREAD

**Issue by sengerts** [rowanwins/shamos-hoey#12]
Hi,

I wanted to use your module to check if a polygonal chain contains self-intersection. However, for some input coordinates, I get the error stated in the title. One example where this error occurs is the following:

``` javascript
const coords = [[[258.75,97.93749809265137],[266.25,96.68749809265137],[272.5,94.18749809265137],[281.25,91.68749809265137],[287.5,89.18749809265137],[301.25,87.93749809265137],[312.5,87.93749809265137],[322.5,87.93749809265137],[332.5,87.93749809265137],[341.25,87.93749809265137],[352.5,90.43749809265137],[362.5,90.43749809265137],[373.75,94.18749809265137],[388.75,95.43749809265137],[401.25,96.68749809265137],[411.25,100.43749809265137],[421.25,100.43749809265137],[443.75,109.18749809265137],[453.75,110.43749809265137],[462.5,111.68749809265137],[471.25,114.18749809265137],[478.75,114.18749809265137],[483.75,115.43749809265137],[491.25,116.68749809265137],[497.5,119.18749809265137],[501.25,119.18749809265137],[503.75,120.43749809265137],[507.5,120.43749809265137],[512.5,120.43749809265137],[516.25,121.68749809265137],[518.75,124.18749809265137],[523.75,125.43749809265137],[528.75,125.43749809265137],[536.25,129.18749809265137],[542.5,130.43749809265137],[547.5,134.18749809265137],[553.75,135.43749809265137],[558.75,135.43749809265137],[563.75,139.18749809265137],[567.5,140.43749809265137],[572.5,144.18749809265137],[576.25,145.43749809265137],[578.75,149.18749809265137],[582.5,150.43749809265137],[587.5,154.18749809265137],[591.25,156.68749809265137],[593.75,160.43749809265137],[598.75,165.43749809265137],[602.5,169.18749809265137],[607.5,171.68749809265137],[608.75,175.43749809265137],[612.5,179.18749809265137],[616.25,181.68749809265137],[617.5,185.43749809265137],[621.25,189.18749809265137],[622.5,190.43749809265137],[622.5,194.18749809265137],[623.75,195.43749809265137],[626.25,196.68749809265137],[626.25,199.18749809265137],[627.5,201.68749809265137],[627.5,205.43749809265137],[628.75,210.43749809265137],[628.75,210.43749809265137],[628.75,214.18749809265137],[627.5,216.68749809265137],[625,220.43749809265137],[621.25,225.43749809265137],[618.75,230.43749809265137],[616.25,235.43749809265137],[611.25,240.43749809265137],[607.5,245.43749809265137],[605,250.43749809265137],[601.25,255.43749809265137],[597.5,260.43749809265137],[595,264.18749809265137],[592.5,265.43749809265137],[591.25,266.68749809265137],[588.75,270.43749809265137],[586.25,271.68749809265137],[583.75,275.43749809265137],[582.5,275.43749809265137],[580,276.68749809265137],[578.75,280.43749809265137],[576.25,280.43749809265137],[572.5,284.18749809265137],[570,285.43749809265137],[568.75,285.43749809265137],[567.5,286.68749809265137],[566.25,289.18749809265137],[565,290.43749809265137],[563.75,290.43749809265137],[561.25,291.68749809265137],[560,294.18749809265137],[557.5,295.43749809265137],[556.25,295.43749809265137],[553.75,296.68749809265137],[551.25,299.18749809265137],[548.75,300.43749809265137],[545,300.43749809265137],[542.5,301.68749809265137],[538.75,304.18749809265137],[535,305.43749809265137],[531.25,305.43749809265137],[528.75,306.68749809265137],[525,309.18749809265137],[521.25,310.43749809265137],[518.75,311.68749809265137],[515,311.68749809265137],[511.25,311.68749809265137],[508.75,314.18749809265137],[506.25,314.18749809265137],[503.75,314.18749809265137],[501.25,314.18749809265137],[498.75,314.18749809265137],[496.25,314.18749809265137],[493.75,314.18749809265137],[492.5,314.18749809265137],[490,314.18749809265137],[488.75,314.18749809265137],[486.25,314.18749809265137],[485,314.18749809265137],[482.5,314.18749809265137],[480,314.18749809265137],[478.75,312.93749809265137],[476.25,312.93749809265137],[475,311.68749809265137],[472.5,310.43749809265137],[471.25,310.43749809265137],[470,309.18749809265137],[468.75,307.93749809265137],[467.5,306.68749809265137],[466.25,306.68749809265137],[465,305.43749809265137],[463.75,305.43749809265137],[461.25,304.18749809265137],[461.25,302.93749809265137],[460,301.68749809265137],[458.75,301.68749809265137],[457.5,300.43749809265137],[456.25,300.43749809265137],[455,299.18749809265137],[453.75,299.18749809265137],[452.5,297.93749809265137],[451.25,297.93749809265137],[450,296.68749809265137],[448.75,295.43749809265137],[447.5,295.43749809265137],[446.25,294.18749809265137],[445,294.18749809265137],[443.75,292.93749809265137],[442.5,292.93749809265137],[441.25,291.68749809265137],[438.75,290.43749809265137],[437.5,290.43749809265137],[436.25,290.43749809265137],[435,289.18749809265137],[433.75,287.93749809265137],[432.5,286.68749809265137],[430,285.43749809265137],[428.75,284.18749809265137],[427.5,282.93749809265137],[426.25,282.93749809265137],[425,281.68749809265137],[423.75,280.43749809265137],[423.75,279.18749809265137],[422.5,277.93749809265137],[421.25,276.68749809265137],[420,276.68749809265137],[418.75,275.43749809265137],[418.75,274.18749809265137],[417.5,272.93749809265137],[416.25,271.68749809265137],[415,271.68749809265137],[415,270.43749809265137],[413.75,270.43749809265137],[413.75,269.18749809265137],[412.5,269.18749809265137],[411.25,269.18749809265137],[411.25,267.93749809265137],[411.25,266.68749809265137],[411.25,265.43749809265137],[410,265.43749809265137],[410,264.18749809265137],[408.75,262.93749809265137],[408.75,261.68749809265137],[407.5,260.43749809265137],[407.5,256.68749809265137],[407.5,254.18749809265137],[406.25,249.18749809265137],[406.25,242.93749809265137],[406.25,237.93749809265137],[406.25,232.93749809265137],[406.25,226.68749809265137],[406.25,222.93749809265137],[406.25,219.18749809265137],[406.25,215.43749809265137],[406.25,212.93749809265137],[406.25,210.43749809265137],[406.25,207.93749809265137],[406.25,205.43749809265137],[407.5,202.93749809265137],[407.5,200.43749809265137],[407.5,199.18749809265137],[407.5,197.93749809265137],[408.75,196.68749809265137],[408.75,195.43749809265137],[408.75,194.18749809265137],[408.75,192.93749809265137],[411.25,192.93749809265137],[412.5,191.68749809265137],[413.75,191.68749809265137],[416.25,190.43749809265137],[417.5,190.43749809265137],[417.5,190.43749809265137]]]
const polygonalChain = {type: 'MultiLineString', coordinates: coords}
console.log(isSimple(polygonalChain))
```

And here is the error message in detail:
``` javascript
/node_modules/shamos-hoey/dist/shamosHoey.esm.js:1067 Uncaught (in promise) TypeError: Cannot read property 'segmentAbove' of null
    at isSimple (/node_modules/shamos-hoey/dist/shamosHoey.esm.js:1067)
    at eval (DrawingCanvas.svelte:85)
    at run (index.mjs:18)
    at Array.map (<anonymous>)
    at index.mjs:1294
    at flush (index.mjs:635)
    at ...
```

I would be very happy if you could help me with that.

Thanks!

**Comment by rowanwins** [comment]
Thanks for the report @sengerts - I'll take a look

**Comment by rowanwins** [comment]
Hi @sengerts - I've just published v`1.0.1` to npm which resolves the issue

**Comment by sengerts** [comment]
Thank you! Will test it out tomorrow

EDIT: Works!
